sql/src/main/java/io/crate/execution/engine/indexing/GroupRowsByShard.java

```diff
@@ -89,15 +85,11 @@ public void accept(ShardedRequests<TReq, TItem> shardedRequests, Row row) {
         for (int i = 0; i < sourceInfoExpressions.size(); i++) {
             sourceInfoExpressions.get(i).setNextRow(row);
         }
-
-        String sourceUriFailure = sourceUriFailureInput.value();
-        if (sourceUriFailure != null) {
-            TItem item = sourceUriFailureFactory.apply(sourceUriFailure);
-            addFailure(shardedRequests, item);
+        if (possibleSourceUriProcessor.apply(shardedRequests)) {
+            // source uri failed processing (reading)
             return;
```

Comment:
If no source uri could be processed, the item is skipped entirely. Is this intended?

Reply:
no it is not. it will only be skipped if there is a sourceUri failure.